src/libcore/mem/maybe_uninit.rs

```diff
@@ -248,6 +248,7 @@ impl<T> MaybeUninit<T> {
     /// [type]: union.MaybeUninit.html
     #[stable(feature = "maybe_uninit", since = "1.36.0")]
     #[inline(always)]
+    #[rustc_promotable]
```

Comment:
Does this have support for feature gating?

Reply:
I don't think so. @oli-obk ?

Reply:
Explicitly puting the `MaybeUninit::uninit()` seems to work. That would make this unnecessary.

https://play.integer32.com/?version=stable&mode=debug&edition=2018&gist=71886586e61fc9a677ce120c825c4aae

Reply:
That playground does not load here? I just get a blank page.

Can you paste the code you are suggesting here in GitHub?

Reply:
Yeah the playground at integer32 broken for me too.

The code was:

```rust
use std::mem::MaybeUninit;

fn main() {
    const A: MaybeUninit<u8> = MaybeUninit::uninit();
    let a_arr = [A; 2];
}
```

But looking at the optimized llvm ir, it seems that llvm emits a `memset` with zero, instead of keeping it uninitialized in case of:

```rust
use std::mem::MaybeUninit;

pub fn abc() -> [MaybeUninit<u8>; 16] {
    const A: MaybeUninit<u8> = MaybeUninit::uninit();
    let a_arr = [A; 16]; //unsafe { std::mem::uninitialized() };
    a_arr
}
```

Reply:
`MaybeUninit<u8>` is `Copy`, so this does not really test the new code paths anyway.

But yes, you can probably "trigger" promotion as usual with a manual `const`.